c331033 김윤진

#데이터 연산

In [39]:
print(f'{2+3= }') 
print(f'{2.0+3= }')
print(f'{2.0+3.0= }')
print(f'{'2'+'3' = }') 

try:
    print(f'{2+'3'= }')
except Exception as e:
    print(e)



2+3= 5
2.0+3= 5.0
2.0+3.0= 5.0
'2'+'3' = '23'
unsupported operand type(s) for +: 'int' and 'str'


In [40]:
print(f'{2-3= }')
print(f'{2.0-3= }') 
print(f'{2.0-3.0= }')
print(f'{2*3= }')
print(f'{2.0*3= }')
print(f'{2.0*3.0= }')
print(f'{2/3 = }') 
print(f'{2.0/3 = }') 
print(f'{2.0/3.0= }') 


2-3= -1
2.0-3= -1.0
2.0-3.0= -1.0
2*3= 6
2.0*3= 6.0
2.0*3.0= 6.0
2/3 = 0.6666666666666666
2.0/3 = 0.6666666666666666
2.0/3.0= 0.6666666666666666


In [41]:
print(f'{20//3  = }')

print(f'{20%3= }')


20//3  = 6
20%3= 2


In [42]:
print(f'{2**3= }')

2**3= 8


In [43]:
var_x = 'Abc'
var_y = [100, 2, 3]

print(f'{var_x + var_x = }')

print(f'{var_y + var_y= }')

print(f'{var_x * 3= }')

print(f'{var_y * 3= }')




var_x + var_x = 'AbcAbc'
var_y + var_y= [100, 2, 3, 100, 2, 3]
var_x * 3= 'AbcAbcAbc'
var_y * 3= [100, 2, 3, 100, 2, 3, 100, 2, 3]


In [44]:
print(f'{ 2==3 = }')
print(f'{ 2!=3 = }')
print(f'{ 2>3 = }')
print(f'{ 2<3 = }')
print(f'{ 2>=3 = }')
print(f'{ 2<=3 = }')

 2==3 = False
 2!=3 = True
 2>3 = False
 2<3 = True
 2>=3 = False
 2<=3 = True


In [45]:
var_y = [1, 2, 3]


print(f'{(var_y == [1, 2, 3]) = }')
print(f'{(var_y == (1, 2, 3)) = }')
print(f'{(var_y == [1, 2]) = }')
print(f'{(var_y != [1, 2]) = }')
print(f'{(var_y > [1, 2]) = }')
print(f'{(var_y < [1, 2]) = }')

(var_y == [1, 2, 3]) = True
(var_y == (1, 2, 3)) = False
(var_y == [1, 2]) = False
(var_y != [1, 2]) = True
(var_y > [1, 2]) = True
(var_y < [1, 2]) = False


In [46]:
True and True

True

In [47]:
var_x = True

var_y = (10 > 20)


print(f'{var_y = }')
print(f'{(var_x and var_y) = }')
print(f'{(var_x or var_y) = }')
print(f'{(not var_x) = }')


var_y = False
(var_x and var_y) = False
(var_x or var_y) = True
(not var_x) = False


In [48]:
print(f'{(2==3) and (2!=3) = }')

(2==3) and (2!=3) = False


In [49]:
var_y = [1, 2, 3]

print(f'{(1 in var_y) = }') 
print(f'{('1' in var_y) = }')
print(f'{(4 in var_y) = }')
print(f'{(1 not in var_y) = }')
print(f'{(4 not in var_y) = }')

(1 in var_y) = True
('1' in var_y) = False
(4 in var_y) = False
(1 not in var_y) = False
(4 not in var_y) = True


In [50]:
var_x = 10

print(f'{(var_x- 10) = }')

print(f'{(var_x is 10) = }')

(var_x- 10) = 0
(var_x is 10) = True


<>:5: SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
<>:5: SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
C:\Users\Public\Documents\ESTsoft\CreatorTemp\ipykernel_8816\642703113.py:5: SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
  print(f'{(var_x is 10) = }')


In [51]:
var_x = 10
var_y = 10.0
var_z= var_x


print(f'{(var_x is var_y) = }')
print(f'{(var_x is var_z) = }')
print(f'{(var_x is not var_y) = }')
print(f'{(var_x is not var_z) = }')





(var_x is var_y) = False
(var_x is var_z) = True
(var_x is not var_y) = True
(var_x is not var_z) = False


In [52]:
var_x = 1

var_x += 1

print(var_x)

2


In [53]:
var_x = 1

var_x += 10; print(var_x)
var_x -= 1; print(var_x)
var_x *= 2; print(var_x)
var_x /= 2; print(var_x)
var_x //= 2; print(var_x)
var_x %= 2; print(var_x)
var_x **= 2; print(var_x)


11
10
20
10.0
5.0
1.0
1.0


In [54]:
print(2**3+5)
print(2**(3+5))
print(2+3 <= 4*5)
print(not 2+3 <= 4*5)
print(not ((2+3) <= (4*5)))


13
256
True
False
False


# 추가학습 내용

In [55]:
sum([1,2])

3

sum은 모든 요소를 더한 값을 반환한다.

iterable 객체에서 나누기의 의미는? [1,2,3]/3은 에러가 난다. iterable 객체에서는 덧셈과 곱셈만 사용 가능하다.

In [56]:
2*5 > 2+5 and not 3*3 > 10

True

식을 계산하면 10 > 7 → True, 9 > 10 → False이다. not False는 True이기에 True and True여서 True이다.

print(f'{ 2+'3' = }')는 에러가 난다. 2는 정수형(int) '3'는 문자열(str) 
정수(int)와 문자열(str)은 + 연산이 불가능하여 TypeError가 발생

print(f'{ 2+ [1,2,3] = }')는 에러가 난다. 2는 정수형(int) [1, 2, 3]는 리스트(list)
정수(int)와 리스트(list)는 + 연산이 불가능하여 TypeError 발생

print(f'{ '123'+ [1,2,3] = }')는 에러가 난다. '123'는 문자열(str) [1, 2, 3]는 리스트(list)
문자열(str)과 리스트(list)는 + 연산이 불가능하여 TypeError 발생.

In [61]:
print(f'{ 2/3 =:.4f}')

 2/3 =0.6667


:.4f는 소수점 4자리까지 표시하라는 뜻이어서 2/3의 결과 0.6666666666666666을 0.6667로 반올림하여 출력함.

In [62]:
list('123') + [1,2,3]

['1', '2', '3', 1, 2, 3]

+ 연산자는 리스트를 연결하는 역할을 함. 두 리스트가 합쳐져서 하나의 리스트가 됨.

In [63]:
'123' + str([1,2,3])

'123[1, 2, 3]'

문자열'123'과 문자열str([1,2,3])을 더하면 문자열이 이어짐

In [66]:
print(f'{ 2!= 3 = }')

 2!= 3 = True


!=는 같지않다는 뜻의 연산자. 2와 3은 다르므로 True 반환됨

In [67]:
print(f'{(2!= 3) = }')

(2!= 3) = True


!=는 같지않다는 뜻의 연산자. 2와 3은 다르므로 True 반환됨

In [71]:
var_y = [1,2,3]
print(f'{var_y > [1,2] = }')

var_y > [1,2] = True


[1,2,3]은 [1,2]보다 요소가 더 많아서 더 큼. (True)

In [72]:
[1,2,3] > [1,2,2]

True

1 == 1, 2 == 2, 3 > 2여서 [1,2,3]가 더 큼.

In [73]:
[1,2,3] > [1,2,2,3,4]

True

1 == 1, 2 == 2, 3 > 2여서 [1,2,3]가 더 큼.

X += a와 X = X+a는 같은 의미임.(복합대입연산자)

In [75]:
10 is 10.0

<>:1: SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
<>:1: SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
C:\Users\Public\Documents\ESTsoft\CreatorTemp\ipykernel_8816\3488692742.py:1: SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
  10 is 10.0


False

정수(int)와 실수(float)는 서로 다른 객체이므로 is 비교 결과는 False

In [ ]:
print(f'{var_y > [1,2] = }')